Archipelago.HollowKnight/IC/Modules/ItemNetworkingModule.cs

```diff
@@ -180,12 +188,18 @@ private async Task Synchronize()
                 NetworkItem seen = session.Items.DequeueItem();
                 Archipelago.Instance.LogDebug($"Fast-forwarding past already-obtained {session.Items.GetItemName(seen.Item)} at index {i}");
             }
+            bool silentGive = !Archipelago.Instance.MenuSettings.AlwaysShowItems;
             // receive from the server any items that are pending
-            while (ReceiveNextItem(true)) { }
-            // ensure any already-checked locations (co-op, restarting save) are marked cleared
+            while (ReceiveNextItem(silentGive)) { }
+            // ensure any already-checked locations (co-op, restarting save, other players collecting) are marked cleared
             foreach (long location in session.Locations.AllLocationsChecked)
             {
-                MarkLocationAsChecked(location, true);
+                // NOTE: this is intentionally always set to silently give during initial sync even when
+                // AlwaysShowItems is true because this shows the items that OTHER players have gotten
+                // FROM this world, which can SERIOUSLY lock up the item display if someone collects a
+                // lot of their items at once (usually via the !collect command). It also prevents these
+                // items from displaying in RecentItemsDisplay (if installed) for the same reason.
+                MarkLocationAsChecked(location, true, true);
```

Comment:
Won't this also hide your own received items from recent items? That seems wrong.